rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -329,24 +337,24 @@ else if (isCached.get(idx) == Location.CACHED) {
             chainedReadRequestChainBuilder.add(new ChainedReadRequestChain().addReadRequestChain(cachedReadRequestChain));
```

Comment:
This was an oversight earlier .. we need not created ChainedReadRequestChain object, just adding cachedReadRequestChain to chainedReadRequestChainBuilder would do